lib/pbench/server/database/database.py

```diff
@@ -12,22 +15,60 @@ class Database:
     db_session = None
 
     @staticmethod
-    def get_engine_uri(config, logger):
-        engine_uri = None
-        try:
-            engine_uri = config.get("database", "uri")
-        except (NoSectionError, NoOptionError) as exc:
-            msg = "Error in configuration file: {}"
-            if logger:
-                logger.error(msg, exc)
-            else:
-                print(msg.format(exc), file=sys.stderr)
-        return engine_uri
+    def get_engine_uri(config):
+        """Convenience method to hide knowledge of the database configuration."""
+        return config.get("database", "uri")
+
+    @staticmethod
+    def wait_for_database(db_uri: str, timeout: int):
+        """Wait for the database server to become available.
+
+        While we encounter "connection refused", sleep one second, and then try
+        again.  No connection attempt is made for a database URI without a host
+        name.
+
+        The timeout argument to `create_connection()` does not play into the
+        retry logic, see:
+
+          https://docs.python.org/3.9/library/socket.html#socket.create_connection
+
+        Arguments:
+
+            timeout : integer number of seconds to wait before giving up
+                      attempts to connect to the database
+
+        Raises a BadConfig exception if the DB URI specifies a host without a
+        port, and the ConnectionRefusedError enountered after the timeout.
```

Comment:
This text looks possibly damaged (and `enountered` is missing its `c`).  Should it be formatted with a `Raises` heading and two entries under it?

Reply:
Fixed.

Reply:
I don't see a new commit for it, but, indeed, it seems to be fixed.